### PR TITLE
test: use and clean up temporary etcd data directory

### PR DESCRIPTION
**Pull request by chahn**
etcd stores its data in the `data-dir`. If this directory is not explicitly specified as a parameter, etcd creates a directory `default.etcd` in the directory where the etcd process is started.

When the Skipper etcd tests begin, the directory `default.etcd` is created by etcd for storing its data but it is not deleted after the etcd process is terminated.
During subsequent tests in the same environment, Skipper would receive routes from previous tests because of the preserved data in etcd and by calling `etcdGetUpdates()` (using `etcdIndex` and `waitIndex`).
This causes the tests to behave unexpectedly.

This patch creates a temporary `data-dir` before starting etcd and makes sure that it is deleted after the tests.
This way the tests are always running with an empty etcd in a defined state.

### CODE REVIEW

File: etcd/etcdtest/mocketcd.go [hunk 1]
```diff
@@ -38,6 +39,15 @@ func randPort() int {
...
 }
 
+// Create a temporary directory for etcd's data-dir.
+func makeTempDir() (string, error) {
+	dir, err := os.MkdirTemp("", "etcdtest")
+	if err != nil {
+		return "", err
+	}
+	return dir, nil
+}
```
Comment (AlexanderYastrebov) [inline etcd/etcdtest/mocketcd.go:49]: This is for single use - lets just inline it.

File: etcd/etcdtest/mocketcd.go [hunk 2]
```diff
@@ -117,6 +135,12 @@ func Stop() error {
 		return nil
 	}
 
+	// Ensure that etcd's data-dir (etcdDataDir) is deleted even if killing the process fails.
+	defer func() {
+		os.RemoveAll(etcdDataDir)
+		etcdDataDir = "" // Clear the dataDir variable.
```
Comment (AlexanderYastrebov) [inline etcd/etcdtest/mocketcd.go:141]: IMO this and other comments are redundant as it is quite clear what is going on.

**Comment by AlexanderYastrebov** [comment]
Thank you, it looks good.

**Comment by AlexanderYastrebov** [comment]
:+1:

File: etcd/etcdtest/mocketcd.go (hunk 2 above)
Comment (chahn) [reply etcd/etcdtest/mocketcd.go]: I stripped the comments as supposed.

File: etcd/etcdtest/mocketcd.go (hunk 1 above)
Comment (chahn) [reply etcd/etcdtest/mocketcd.go]: It is inline now with the latest commit.
Thanks for the feedback @AlexanderYastrebov

**Comment by MustafaSaber** [comment]
👍

**Comment by chahn** [comment]
Thanks for merging and especially for your support @AlexanderYastrebov and @MustafaSaber

**Pull request merged**